atom/browser/api/atom_api_web_contents.cc

```diff
@@ -2860,6 +2860,25 @@ void WebContents::OnTabCreated(const mate::Dictionary& options,
     tab_helper->SetPinned(pinned);
   }
 
+  int window_id = -1;
+  ::Browser *browser = nullptr;
+  if (options.Get("windowId", &window_id) && window_id != -1) {
+    tab_helper->SetWindowId(window_id);
+    auto api_window =
+        mate::TrackableObject<Window>::FromWeakMapID(isolate(), window_id);
+    if (api_window)
+      browser = api_window->window()->browser();
+  }
+  if (!browser) {
+    browser = owner_window()->browser();
+  }
+
+  // This call needs to happen before both tab_helper->SetAutoDiscardable()
```

Comment:
do you have steps to replicate this? I don't think this is the fix we want

Reply:
We have an updated commit. `SetAutoDiscardable()` and `Discard()` are moved below the call to `AddNewContents()` and only called if `!was_blocked`,  meaning that `SetBrowser()` was called there.

Reply:
New version is https://github.com/brave/muon/commit/8868bc4b70533f728aa1543db60f5100772ca2fc